packages/ui/src/modal/sweep/SweepModalRenderer.tsx

```diff
@@ -0,0 +1,590 @@
+          total += token.market.floorAsk.price.amount.decimal
+        }
+      }
+
+      return total
+    }, 0)
+
+    setTotal(total)
+  }, [selectedTokens, isChainCurrency])
+
+  // sort tokens by price
+  const sortByPrice = useCallback((a: Token, b: Token) => {
+    const aPrice = a.market?.floorAsk?.price?.amount?.decimal
+    const bPrice = b.market?.floorAsk?.price?.amount?.decimal
+
+    if (aPrice === undefined) {
+      return 1
+    } else if (bPrice === undefined) {
+      return -1
+    } else {
+      return aPrice - bPrice
+    }
+  }, [])
+
+  const updateSelectedTokens = useCallback(
+    (tokens: Token[], maxTokens: number) => {
+      let pools: { [poolId: string]: number } = {}
+      let updatedTokens: Token[] = []
+
+      // Create a copy of the availableTokens
+      let processedTokens = [...tokens]
+
+      for (let i = 0; i < maxTokens && i < processedTokens.length; i++) {
+        const token = processedTokens[i]
+
+        updatedTokens.push(token)
+
+        // handle if token is in a dynamic pricing pool
+        if (
+          token.market?.floorAsk?.dynamicPricing?.kind === 'pool' &&
+          token?.market?.floorAsk?.dynamicPricing?.data?.pool &&
+          token?.market?.floorAsk?.dynamicPricing?.data?.prices
+        ) {
+          const poolId = token.market.floorAsk.dynamicPricing.data
+            .pool as string
+          const poolPrices = token.market.floorAsk.dynamicPricing.data
+            .prices as FloorAskPrice[]
+
+          // update the pools
+          if (pools[poolId] === undefined) {
+            pools[poolId] = 1
+          } else {
+            pools[poolId] += 1
+          }
+
+          // update the prices of other tokens in the same pool
+          processedTokens = processedTokens.map((processedToken) => {
+            if (
+              processedToken.market?.floorAsk?.dynamicPricing?.data?.pool ===
+                poolId &&
+              !updatedTokens.some(
+                (updatedToken) =>
+                  updatedToken.token?.tokenId === processedToken.token?.tokenId
+              )
+            ) {
+              if (pools[poolId] < poolPrices.length) {
+                processedToken.market.floorAsk.price = poolPrices[pools[poolId]]
+              } else {
+                processedToken.market.floorAsk.price = undefined
+              }
+            }
+            return processedToken
+          })
+
+          // sort tokens with the updated prices
+          processedTokens.sort(sortByPrice)
+        }
+      }
+
+      return updatedTokens
+    },
+    [sortByPrice]
+  )
+
+  // Add by item
+  useEffect(() => {
+    const updatedTokens = updateSelectedTokens(availableTokens, itemAmount || 0)
+    setSelectedTokens(updatedTokens)
+  }, [itemAmount, updateSelectedTokens])
+
+  // Add by price
+  useEffect(() => {
+    const maxTokens = availableTokens.reduce(
+      (count, token) => {
+        const tokenPrice = isChainCurrency
```

Comment:
This won't take into account dynamic pools because the price moves along the curve, so you would need to know exactly what the price would be depending on the items selected from that pool